### DISCUSSION TITLE
InvalidRequestError: Unknown PG numeric type when using postgres pgvector extension

### DISCUSSION THREAD

**Discussion by saeedesmaili** [fastapi/sqlmodel#580] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pgvector.sqlalchemy import Vector

## model
class Record(SQLModel, table=True):
    id: UUID = Field(default_factory=uuid4, primary_key=True)
    text: str = Field(default=None)
    start: int = Field(default=None)
    token_count: int = Field(default=None)
    embedding: List[float] = Field(default=None, sa_column=Vector(1536))

## controllers
def get_record(session: Session, id: UUID):
    return session.get(Record, id)

def create_or_update_record(session: Session, record: RecordCreate):
    db_record = session.exec(
        select(Record).where(Record.id == record.id)
    ).first()
    # rest of the function
```

### Description

I'm using [pgvector](https://github.com/pgvector/pgvector-python) to insert vectors to the database. But when I need to query the table (for updating a row for example), I get the following error:

> api_1        |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/fastapi.py", line 84, in _sentry_call
> api_1        |     return old_call(*args, **kwargs)
> api_1        |            ^^^^^^^^^^^^^^^^^^^^^^^^^
> api_1        |   File "/usr/src/app/app/api/controllers/records.py", line 119, in create_or_update_record_endpoint
> api_1        |     return create_or_update_record(session, record)
> api_1        |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> api_1        |   File "/usr/src/app/app/api/controllers/records.py", line 53, in create_or_update_record
> api_1        |     db_record = session.exec(
> api_1        |                     ^^^^^^^^^^^^^
> api_1        |   File "/usr/local/lib/python3.11/site-packages/sqlmodel/orm/session.py", line 60, in exec
> api_1        |     results = super().execute(
> api_1        |               ^^^^^^^^^^^^^^^^
> rest of the error ...
> api_1        |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/sql/type_api.py", line 702, in _cached_result_processor
> api_1        |     d[coltype] = rp = d["impl"].result_processor(dialect, coltype)
> api_1        |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> api_1        |   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 525, in result_processor
> api_1        |     raise exc.InvalidRequestError(
> api_1        | sqlalchemy.exc.InvalidRequestError: Unknown PG numeric type: 24664

I had the issue with inserting vectors as well, but I fixed it by changing from:

```python
def create_record(session: Session, record: RecordCreate):
    new_record = Record(**record.dict())
    session.add(new_record)
    session.commit()
    session.refresh(new_record)
    return new_record
```

to:

```python
def create_record(session: Session, record: RecordCreate):
    new_record = Record(**record.dict())
    session.add(new_record)
    session.commit()
    # session.refresh(new_record)
    return dict(new_record) 
```

Any idea how can I fix this?

### Operating System

macOS

### SQLModel Version

0.0.8

### Python Version

3.9.13

**Comment by sarahwooders** [comment]
Was this ever resolved? I am also using pgvector.

**Reply by saeedesmaili** [reply, marked as answer]
Yeah I fixed it by changing:
```python
embedding: List[float] = Field(default=None, sa_column=Vector(1536))
```
to:
```python
embedding: List[float] = Field(default=None, sa_column=Column(Vector(1536)))
```

**Comment by digitalfuturo** [comment]
Thanks a lot! This helped
